src-docs/irc_bridge.py.md

```diff
@@ -0,0 +1,89 @@
+<!-- markdownlint-disable -->
+
+<a href="../src/irc_bridge.py#L0"><img align="right" style="float:right;" src="https://img.shields.io/badge/-source-cccccc?style=flat-square"></a>
+
+# <kbd>module</kbd> `irc_bridge.py`
+Provide the IRC bridge class to represent the matrix-appservice-app plugin for Synapse. 
+
+**Global Variables**
+---------------
+- **IRC_SERVICE_NAME**
+
+
+---
+
+## <kbd>class</kbd> `IRCBridge`
+A class representing the IRC bridge plugin for Synapse application. 
+
+See https://github.com/matrix-org/matrix-appservice-irc/ for more details about it. 
+
+<a href="../src/irc_bridge.py#L41"><img align="right" style="float:right;" src="https://img.shields.io/badge/-source-cccccc?style=flat-square"></a>
+
+### <kbd>function</kbd> `__init__`
+
+```python
+__init__(charm: CharmBase, charm_state: CharmState)
```

Comment:
Is this an observer? If not, what is the reason for needing the `charm` object please?

Reply:
I think it is intended to be an observer, perhaps include that in the name of the class?

Reply:
Correct, done.